flocker/volume/filesystems/zfs.py

```diff
@@ -67,26 +71,100 @@ def connectionLost(self, reason):
         del self._result
 
 
-def zfs_command(reactor, arguments):
+class _AsyncLZC(object):
```

Comment:
I know it's early days, but a doc string here would help me understand the design of this wrapper.

Instead of wrapping every libzfs_core function with `_async_wrapper` consider creating an `ILibZFSCore` Zope Interface and using `auto_threaded` from https://github.com/ClusterHQ/flocker/blob/master/flocker/common/_thread.py

Or perhaps you'll have to create a slightly higher level abstraction in order to use that. eg `IZFSClient` which is wrapped to give `IZFSClientAsync`. That's how @exarkun organised the `IBlockDeviceAPI` .